torch/_inductor/codegen/cpp.py

```diff
@@ -1995,43 +1996,34 @@ def index_expr(expr, dtype):
                     self.disable_vec(f"index_expr: {expr}, dtype {dtype}")
                     return self.cse.newvar()
 
-                def mod_indexing_rep(x, y, z):
-                    if z.is_constant():
-                        return x / y
-
-                    # never really happens, we'll bail on optimizing
-                    return (x / y) % z
-
-                def indexing_div_rep(x, y):
-                    return x / y
+                def can_use_int32():
+                    free_symbols = list(expr.free_symbols)
+                    vars_ranges = {
+                        k: ValueRanges(0, v)
+                        for k, v in zip(self.itervars, self.ranges)
+                        if k in free_symbols
+                    }
+                    if not vars_ranges or len(vars_ranges) != len(free_symbols):
```

Comment:
Is this equivalent to `any(x.startwith("tmp") for x in free_symbols)`? In other words, here we are asking whether the expression has indirect indexing, as we don't have bounds at hand for those here. 
If that's the case, using this other more explicit condition and leaving a comment would help to understand the logic here.

Reply:
I don't think it is equivalent to ```any(x.startwith("tmp") for x in free_symbols)```. There has a case that the expr is just kernel input(```s0```), which is not related to ```itervars``` and it is not buffer indexing.

Reply:
Right, then same question with `startswith("tmp") or startswith("s")`. It'd be nice to know more clearly which cases we know how to treat and which we don't.

Reply:
Yes, I agree with you that knowing more clearly what cases have is better. Let me add this as a TODO work.